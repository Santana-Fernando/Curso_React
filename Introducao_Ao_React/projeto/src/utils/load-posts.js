export const loadPosts = async () => {
    let posts = {}
    //const postsResponse = fetch('https://jsonplaceholder.typicode.com/todos')
    await fetch('https://jsonplaceholder.typicode.com/todos')
    .then(response => response.json())
    .then(json => posts = json)

    return posts
}