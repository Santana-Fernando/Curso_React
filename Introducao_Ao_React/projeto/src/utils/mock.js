export const mock = {
    post: {
        id: 'Ola', 
        title: 'Ola meu nome e fernando'
    }
}