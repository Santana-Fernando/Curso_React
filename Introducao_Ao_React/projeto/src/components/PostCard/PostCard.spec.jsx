import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { PostCard } from '.';
import { mock } from '../../utils/mock';

describe('<PostCard />', () => {

    it('should render PostCard correctly', () => {
        render(<PostCard {...mock}/>)
        //expect.assertions(2)
    

        const postCardH1 = screen.getByRole('heading', {name: 'Ola'})
        const postCardP = screen.getByText('Ola meu nome e fernando')

        expect(postCardH1).toBeInTheDocument()
        expect(postCardP).toBeInTheDocument()
    });

    it('should match snapshot', () => {
        const {container} = render(<PostCard {...mock}/>)
        expect(container.firstChild).toMatchSnapshot()
    });


});