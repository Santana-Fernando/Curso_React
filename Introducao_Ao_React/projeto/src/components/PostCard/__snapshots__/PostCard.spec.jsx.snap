// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <div
    class="post-content"
  >
    <h1>
      Ola
    </h1>
    <p>
      Ola meu nome e fernando
    </p>
  </div>
</div>
`;
